static/js/hotkey.js

```diff
@@ -532,6 +544,23 @@ exports.process_hotkey = function (e, hotkey) {
             return true;
         }
 
+        if (event_name === "narrow_compose" && compose_state.composing()) {
+            if (compose_state.get_message_type() === 'stream' && compose_state.stream_name() !== '') {
```

Comment:
Can we move most of this code into a function called `narrow.to_compose_target`?  We try to minimize the amount of code in `hotkey.js` so that it only dispatches stuff.